File: src/test/java/pageobjects/MainPageSectionsTest.java
package pageobjects;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;

public class MainPageSectionsTest {
    private static WebDriver driver ;
    private BurgerMainPage objBurgerMainPage;

    @Before
    public void setUp(){
        driver =new ChromeDriver();
        driver.get("https://stellarburgers.nomoreparties.site/");
        objBurgerMainPage = new BurgerMainPage(driver);
    }
    @Test
    //All product sections are available on main page
    public void testMainPageSections() {
        int tmp = objBurgerMainPage.identifySection();
        for(int i = 0; i<objBurgerMainPage.getSectionsQuantity();i++){
            if (i != tmp){
                objBurgerMainPage.waitForNSectionVisibility(i);
                Assert.assertTrue(driver.findElement(objBurgerMainPage.returnNSection(i)).isDisplayed());
            }
        }
        objBurgerMainPage.waitForNSectionVisibility(tmp);
        Assert.assertTrue(driver.findElement(objBurgerMainPage.returnNSection(tmp)).isDisplayed());
    }
    @After
    public void tearDown(){
        driver.quit();
    }
}

File: src/test/java/pageobjects/MainPageSectionsTest.kt
package pageobjects

import org.junit.After
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.openqa.selenium.WebDriver
import org.openqa.selenium.chrome.ChromeDriver

class MainPageSectionsTest {
    private lateinit var driver: WebDriver
    private lateinit var page: BurgerMainPage
    @Before
    fun setUp() {
        driver = ChromeDriver()
        driver.get("https://stellarburgers.nomoreparties.site/")
        page = BurgerMainPage(driver)
    }

    @Test //All product sections are available on main page
    fun testMainPageSections() {
        val tmp: Int = page.identifySection()
        for (i in 0 until page.sectionsQuantity) {
            if (i != tmp) {
                page.waitForNSectionVisibility(i)
                assertTrue(driver.findElement(page.returnNSection(i)).isDisplayed)
            }
        }
        page.waitForNSectionVisibility(tmp)
        assertTrue(driver.findElement(page.returnNSection(tmp)).isDisplayed)
    }
    @After
    fun tearDown() {
        driver.quit()
    }
}